File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/cruising/CruisingComponent.java
package io.github.sithengineer.motoqueiro.ui.cruising;

import dagger.Subcomponent;
import io.github.sithengineer.motoqueiro.hardware.bluetooth.BluetoothModule;
import io.github.sithengineer.motoqueiro.hardware.bluetooth.MiBandModule;
import io.github.sithengineer.motoqueiro.scope.ActivityScope;

@ActivityScope
@Subcomponent(modules = { CruisingModule.class, MiBandModule.class, BluetoothModule.class })
public interface CruisingComponent {
  void inject(CruisingFragment fragment);
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/cruising/CruisingComponent.kt
package io.github.sithengineer.motoqueiro.ui.cruising

import dagger.Subcomponent
import io.github.sithengineer.motoqueiro.hardware.bluetooth.BluetoothModule
import io.github.sithengineer.motoqueiro.hardware.bluetooth.MiBandModule
import io.github.sithengineer.motoqueiro.scope.ActivityScope

@ActivityScope
@Subcomponent(modules = [CruisingModule::class, MiBandModule::class, BluetoothModule::class])
interface CruisingComponent {
  fun inject(fragment: CruisingFragment)
}
